metrics_sdk/test/opentelemetry/sdk/metrics/aggregation/aggregation_temporality_test.rb

```diff
@@ -0,0 +1,295 @@
+      it 'returns delta when aggregation_temporality is :delta' do
+        result = OpenTelemetry::SDK::Metrics::Aggregation::AggregationTemporality.determine_temporality(
+          aggregation_temporality: :delta,
+          instrument_kind: :counter,
+          default: :cumulative
+        )
+        _(result.temporality).must_equal :delta
+        _(result.delta?).must_equal true
+      end
+
+      it 'returns cumulative when aggregation_temporality is :cumulative' do
+        result = OpenTelemetry::SDK::Metrics::Aggregation::AggregationTemporality.determine_temporality(
+          aggregation_temporality: :cumulative,
+          instrument_kind: :counter,
+          default: :delta
+        )
+        _(result.temporality).must_equal :cumulative
+        _(result.cumulative?).must_equal true
+      end
+    end
+
+    describe 'with case variations' do
+      it 'handles uppercase DELTA' do
+        result = OpenTelemetry::SDK::Metrics::Aggregation::AggregationTemporality.determine_temporality(
+          aggregation_temporality: 'DELTA',
+          instrument_kind: :counter,
+          default: :cumulative
+        )
+        _(result.temporality).must_equal :delta
+      end
+
+      it 'handles uppercase CUMULATIVE' do
+        result = OpenTelemetry::SDK::Metrics::Aggregation::AggregationTemporality.determine_temporality(
+          aggregation_temporality: 'CUMULATIVE',
+          instrument_kind: :counter,
+          default: :delta
+        )
+        _(result.temporality).must_equal :cumulative
+      end
+
+      it 'handles uppercase LOWMEMORY' do
+        result = OpenTelemetry::SDK::Metrics::Aggregation::AggregationTemporality.determine_temporality(
+          aggregation_temporality: 'LOWMEMORY',
+          instrument_kind: :counter,
+          default: :cumulative
+        )
+        _(result.temporality).must_equal :delta
+      end
+
+      it 'handles lowercase delta' do
+        result = OpenTelemetry::SDK::Metrics::Aggregation::AggregationTemporality.determine_temporality(
+          aggregation_temporality: 'delta',
+          instrument_kind: :counter,
+          default: :cumulative
+        )
+        _(result.temporality).must_equal :delta
+      end
+
+      it 'handles lowercase cumulative' do
+        result = OpenTelemetry::SDK::Metrics::Aggregation::AggregationTemporality.determine_temporality(
+          aggregation_temporality: 'cumulative',
+          instrument_kind: :counter,
+          default: :delta
+        )
+        _(result.temporality).must_equal :cumulative
+      end
+
+      it 'handles lowercase lowmemory' do
+        result = OpenTelemetry::SDK::Metrics::Aggregation::AggregationTemporality.determine_temporality(
+          aggregation_temporality: 'lowmemory',
+          instrument_kind: :counter,
+          default: :cumulative
+        )
+        _(result.temporality).must_equal :delta
+      end
+    end
+
+    describe 'with unknown string values' do
```

Comment:
What do you think about adding a log message to let people know when the string value doesn't match? I could see myself writing `low_memory` instead of `lowmemory` or making some other typo that would be hard to track down.